shared/js/l10n.js

```diff
@@ -1371,17 +1387,59 @@
     this._emitter.emit('ready');
   }
 
-  Context.prototype.registerLocales = function(defLocale, available) {
-    /* jshint boss:true */
-    this.availableLocales = [this.defaultLocale = defLocale];
+  function getMatchingLangpack(lpVersions) {
+    for (var i in lpVersions) {
+      if (lpVersions[i].target === navigator.mozL10n.gaiaVersion) {
+        return lpVersions[i];
+      }
+    }
+    return null;
+  }
 
-    if (available) {
-      for (var i = 0, loc; loc = available[i]; i++) {
-        if (this.availableLocales.indexOf(loc) === -1) {
-          this.availableLocales.push(loc);
+  function extendLanguagesList(availableLangs, extraLangs) {
```

Comment:
Could you add a comment with examples of `availableLangs` and `extraLangs` and explain why they're of different types, please?